### PR TITLE
Update HttpLoggingAttribute API

### LINKED ISSUE #43222: Make Http Logging Middleware Endpoint aware
## Background and Motivation
I need to log the body of every requests that are coming to the application, but I want to exclude some endpoints. For example I don't want to log the body of the login methods. I suggest adding metadata for enabling and disabling log for http logging middleware.

The new behavior would be like these: if endpoint has `DisableLoggingAttribute` the middleware would be skipped completely. when `EnableLoggingAttribute` is available the middleware use its `LoggingFields` to log the request/response and if there are no metadata available, it would use `HttpLoggingOptions.LoggingFields`.

Also should other `HttpLoggingOptions` properties be available in the metadata or is logging fields enough?

## Proposed API

```diff
namespace Microsoft.AspNetCore.HttpLogging;

+ public class EnableLoggingAttribute : Attribute
+{
+    public EnableLoggingAttribute(HttpLoggingFields fields) { }
+    public HttpLoggingFields LoggingFields { get; }
+}

+ public class DisableLoggingAttribute : Attribute
+{
+}

namespace Microsoft.AspNetCore.Builder;

+public static class HttpLoggingEndpointConventionBuilderExtensions
+{
+    public static TBuilder EnableLogging<TBuilder>(this TBuilder builder, HttpLoggingFields loggingFields) where TBuilder : IEndpointConventionBuilder;
+    public static TBuilder DisableLogging<TBuilder>(this TBuilder builder) where TBuilder : IEndpointConventionBuilder;
+}
```

## Usage Examples
```csharp
app.MapPost("/login", ... ).DisableLogging();
app.MapPost("/uploadFile", ... ).EnableLogging(HttpLoggingFields.RequestPropertiesAndHeaders);
```

### PR DESCRIPTION
**Pull request by BrennanConroy**
Closes https://github.com/dotnet/aspnetcore/issues/43222

### CODE REVIEW

File: src/Middleware/HttpLogging/src/HttpLoggingAttribute.cs [hunk 1]
```diff
@@ -13,32 +13,73 @@ public sealed class HttpLoggingAttribute : Attribute
...
+    /// <exception cref="InvalidOperationException">Thrown when getting <see cref="ResponseBodyLogLimit"/> if it hasn't been set to a value. Check <see cref="IsResponseBodyLogLimitSet"/> first.</exception>
+    public int ResponseBodyLogLimit
+    {
+        get
+        {
+            if (IsResponseBodyLogLimitSet)
+            {
+                return _responseBodyLogLimit;
+            }
+            throw new InvalidOperationException("TODO");
```
Comment (Tratcher) [inline src/Middleware/HttpLogging/src/HttpLoggingAttribute.cs:76]: Lol

**Review by Tratcher — APPROVED** [review]

File: src/Middleware/HttpLogging/src/HttpLoggingAttribute.cs (hunk 1 above)
Comment (BrennanConroy) [reply src/Middleware/HttpLogging/src/HttpLoggingAttribute.cs:76]: Oh no 😆

Comment (BrennanConroy) [inline src/Middleware/HttpLogging/src/HttpLoggingAttribute.cs:76]:
```suggestion
            throw new InvalidOperationException($"{nameof(ResponseBodyLogLimit)} was not set. Check {nameof(IsResponseBodyLogLimitSet)} before accessing this property.");
```

File: src/Middleware/HttpLogging/src/HttpLoggingAttribute.cs [hunk 2]
```diff
@@ -13,32 +13,73 @@ public sealed class HttpLoggingAttribute : Attribute
...
+    public int RequestBodyLogLimit
+    {
+        get
+        {
+            if (IsRequestBodyLogLimitSet)
+            {
+                return _requestBodyLogLimit;
+            }
+
+            throw new InvalidOperationException("TODO");
```
Comment (BrennanConroy) [inline src/Middleware/HttpLogging/src/HttpLoggingAttribute.cs:48]:
```suggestion
            throw new InvalidOperationException($"{nameof(RequestBodyLogLimit)} was not set. Check {nameof(IsRequestBodyLogLimitSet)} before accessing this property.");
```

**Pull request merged**